model/registry/instrumentation.yaml

```diff
@@ -0,0 +1,24 @@
+groups:
+- id: registry.instrumentation
+  prefix: instrumentation
+  type: attribute_group
+  brief: >
+    Describes properties about the library that is being instrumented or that
+    is responsible for performing the instrumentation.
+  attributes:
+    - id: source.repo
+      type: string
+      stability: experimental
+      brief: The repository associated with the InstrumentationScope.
+      examples: [github.com/organization/repo, gitlab.com/group/repo]
+    - id: source.target
```

Comment:
See my comment below - I'm not sure why build target would be interesting to end users and would prefer to see an artifact name here.

I'd suggest `instrumentation.scope|library.artifact_name`

Reply:
I believe I addressed that question here:

 - https://github.com/open-telemetry/semantic-conventions/pull/1047#discussion_r1619399324

"the build target can be relevant if the same repo contains multiple different variants of the same thing. For example, you could imagine that there is fast_debug_build, asan_enabled_build, optimized_deployment_build of the component. If there are unusual behaviors or problems that are happening, knowing which of these variants was used could provide some insight regarding what is going on"

The "artifact" or "asset" implies some kind of bundling/packaging/distribution, which may not always be applicable. But it may still be useful for other contexts.

I'm going to resolve this here on this PR, because these changes have been extracted to:

 - https://github.com/open-telemetry/semantic-conventions/pull/1094

I've narrowed the scope of _this_ pull request to just `gcp.client.service` under the philosophy that smaller, incremental, more narrowly focused changes are easier to discuss, review, and update compared to large ones.